### ISSUE TITLE
Cannot duplicate structure fields more than once in a row.

### ISSUE THREAD

**Issue by agatti** [NationalSecurityAgency/ghidra#4229]
**Describe the bug**
It is not possible to duplicate a structure fields more than once in a row using keyboard shortcuts.

**To Reproduce**
1. Open a file in the CodeBrowser
2. Create a new structure
3. Create a field by clicking on the green "plus" toolbar button, change its data type and/or give it a name
4. Duplicate the field using the appropriate keyboard shortcut (Option-D on macOS) 
5. Trigger the shortcut again.
6. No further field duplications happen.

**Expected behavior**
Fields can be duplicated multiple times in a row.

**Screenshots**
Here's an animated GIF that shows the problem (I've added an overlay with keys being pressed)

![output](https://user-images.githubusercontent.com/235740/166853605-703b624b-2380-4bc7-9746-3737a2de174b.gif)

**Attachments**
If applicable, please attach any files that caused problems or log files generated by the software.

**Environment (please complete the following information):**
 - OS: macOS 12.3.1
 - Java Version: 11.0.15
 - Ghidra Version: 10.1.3
 - Ghidra Origin: GitHub binary release

**Comment by Swyter** [comment]
Yeah, structure/enum editing is something that feels unnecessarily stilted. While the layout offset/packing visualization comes in handy, it would be neat to have a C-struct text mode editor that makes it easier to copy-paste and duplicate chunks between elements and in and out of the program.

A bit like what we already have while editing functions directly.

Another idea would be making _hex view_ mode in the editor persistent, I find myself toggling it on all the time.

**Comment by ghidra1** [comment]
We have talked about adding a free-form C structure entry mechanism.  Although, when reverse engineering a structure (i.e., packing disabled - default mode) such an editor can not be used since component placement must be very surgical and may be sparse - known vs unknown components).

**Comment by ghidra1** [comment]
When packing is disabled (default mode) components may only be placed at the end of the structure or where undefined (i.e., padding) components exist.  Provided this is the case one change we should probably make is to **select the newly added component** allowing the duplicate action to be repeated provided there is space.
